lxd/auth/authorization.go

```diff
@@ -45,16 +95,38 @@ type UserAccess struct {
 	Projects map[string][]string
 }
 
-func LoadAuthorizer(name string, config map[string]any, logger logger.Logger, projectsGetFunc func() (map[int64]string, error)) (Authorizer, error) {
-	driverFunc, ok := authorizers[name]
+type Opts struct {
+	config          map[string]any
+	projectsGetFunc func(ctx context.Context) (map[int64]string, error)
```

Comment:
Could we have a comment on this as to what its for, im not seeing why each driver needs its own project function?

Reply:
I've added a comment above the `WithProjectsGetFunc` option function.